lib/winapi/xapi.c

```diff
@@ -5,9 +5,44 @@
  */
 
 #include <xapi.h>
+#include <assert.h>
 #include <stdio.h>
+#include <string.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+__attribute__((constructor)) static VOID setup_d_drive (VOID)
+{
+    NTSTATUS status;
+    HANDLE handle;
+    ANSI_STRING driveString = RTL_CONSTANT_STRING("\\??\\D:");
+    OBJECT_ATTRIBUTES objattr;
+
+    InitializeObjectAttributes(&objattr, &driveString, OBJ_CASE_INSENSITIVE, NULL, NULL);
+    status = NtOpenSymbolicLinkObject(&handle, &objattr);
+    if (NT_SUCCESS(status)) {
+        NtClose(handle);
+        return;
+    }
+
+    // D: doesn't exist yet, so we create it
+    CHAR targetPath[MAX_PATH];
+
+    // Retrieve the XBE path
+    strncpy(targetPath, XeImageFileName->Buffer, XeImageFileName->Length);
+    targetPath[XeImageFileName->Length] = '\0';
+
+    // Cut off the XBE file name by inserting a null-terminator
+    PCHAR filenameStr;
+    filenameStr = strrchr(targetPath, '\\');
+    assert(filenameStr != NULL);
+    *(filenameStr + 1) = '\0';
```

Comment:
This retrieval of the XBE path should ideally be a separate function as we might want to re-use that in the future?
Or more generally, some `dirname`-like function?

Reply:
I moved it do a `XGetXbePath` function now.